crates/sozo/src/ops/events.rs

```diff
@@ -29,7 +38,120 @@ pub async fn execute(args: EventsArgs, env_metadata: Option<Environment>) -> Res
 
     let res = provider.get_events(event_filter, continuation_token, chunk_size).await?;
 
-    let value = serde_json::to_value(res)?;
-    println!("{}", serde_json::to_string_pretty(&value)?);
+    if json {
+        let value = serde_json::to_value(res)?;
+        println!("{}", serde_json::to_string_pretty(&value)?);
+    } else {
+        parse_and_print_events(res, events_map);
+    }
     Ok(())
 }
+
+fn parse_and_print_events(
+    res: starknet::core::types::EventsPage,
+    events_map: HashMap<String, Vec<Event>>,
+) {
+    println!("Continuation token: {:?}", res.continuation_token);
+    println!("----------------------------------------------");
+    for event in res.events {
+        if let Some(e) = parse_event(event.clone(), &events_map) {
+            println!("{}\n", e);
+        } else {
+            // Couldn't parse event
+            println!("{}\n", serde_json::to_string_pretty(&event).unwrap());
```

Comment:
Can we update to use the logging framework we use in other tools? The scarb ui one?

Reply:
we can but it wouldn't be very useful in this case, because of the way this parsing works we don't know until very end of parsing that if a event can be parsed. so we cannot print information in steps like migration ui. would need to pass the whole string to scarb, so no formatting can be done there.

one thing i would think of after we finalize how we want to print events, we can create an `AnyEvent` type which is contains `vec<Item>`  to be printed similar to how cairo compiler handles it. But I am not sure if that would be worth the effort.

Reply:
ok sounds good. let's be sure to track it and land it later. ideally we have a reusable api for event parsing, so we can include it in katana as well and log the parsed event logs during development when trace logging is enabled